lib/logstash/filters/mutate.rb

```diff
@@ -342,11 +342,17 @@ def gsub_dynamic_fields(event, original, needle, replacement)
   def uppercase(event)
     @uppercase.each do |field|
       original = event[field]
+      # in certain cases JRuby returns a proxy wrapper of the event[field] value
+      # therefore we can't assume that we are modifying the actual value behind
+      # the key so read, modify and overwrite
       event[field] = case original
         when Array
-          original.map(&:upcase!)
+          # can't map upcase! as it replaces an already upcase value with nil
+          # ["ABCDEF"].map(&:upcase!) => [nil]
```

Comment:
well, a-priori I would never tend to use a (`String`) bang method in a functional style like with `map` since the bang methods are meant to modify the object in-place and the actual method return value can be anything and have weird semantics unlike a "proper" non-bang method function with applies to a `String` and return a `String`.